mock/main.js

```diff
@@ -1,13 +1,25 @@
 require('./migration');
 
 const express = require('express');
+const cors = require('cors');
+
 const routers = require('./routes');
 
 const app = express();
+
+
 const port = 3000;
 
+app.use(cors());
+
 app.use(express.json());
 
+app.use(cors({
+  origin: 'http://localhost:8080',
+  credentials: true,
+  optionsSuccessStatus: 200 
```

Comment:
このcorsってライブラリ使ったこと無いので、教えてください！
この2つのオプションをつけることで、どうなります？？
```
credentials: true,
optionsSuccessStatus: 200 
```

Reply:
credentials: trueは Cookieや HTTP認証による認証情報を渡す場合、これがないとレスポンスを捨ててしまうそうです。
つまりユーザーのログイン情報などのデータを保持するわけではないので必要ないのでしょうか？

optionsSuccessStatus: 200 はレガシーブラウザのサポートになるそうですこちらは必要ないと思われますので消しておきます

Reply:
> credentials: Configures the Access-Control-Allow-Credentials CORS header. Set to true to pass the header, otherwise it is omitted.

http://expressjs.com/en/resources/middleware/cors.html

> Access-Control-Allow-Credentials レスポンスヘッダーは、リクエストの資格情報モード (Request.credentials) が include である場合に、レスポンスをフロントエンドの JavaScript コードに公開するかどうかをブラウザーに指示します。

> 証明書の資格情報モード (Request.credentials) が include である場合、レスポンスがフロントエンドの JavaScript コードに公開されるのは Access-Control-Allow-Credentials の値が true である場合のみです。

https://developer.mozilla.org/ja/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials

Request.credentialsがincludeの場合の話みたいですね

> Request インターフェースの credentials 読み取り専用プロパティは、クロスオリジンリクエストの場合、ユーザーエージェントが ほかのドメインからクッキーを送信すべきかどうかを示します。これは XHR の withCredentials フラグと似ていますが、（2 つではなく）3 つの値があります：

> omit：決してクッキーを送信しない。
same-origin：URL が呼び出し元のスクリプトと同一オリジンだった場合のみ、クッキーを送信する。
include：クロスオリジンの呼び出しであっても、常にクッキーを送信する。

https://developer.mozilla.org/ja/docs/Web/API/Request/credentials

今回のシステムの場合は、
Request.credentialsはsame-originであるべきなので、
credentials: trueの記述は削除してください！

Reply:
optionsSuccessStatus: 200も削除してください！

Reply:
あー、開発時は同一オリジンにならないのかあ。。。

Reply:
vue.config.jsに
```
devServer: {
  proxy: {
    "^/api": {
      target: "http://localhost:3000"
    }
  }
},
```

https://cli.vuejs.org/config/#devserver-proxy

上記を記載して、corsの設定を削除したら動きます？

Reply:
どのみに、今回はcookieを使っていないので、削除してください！

Reply:
ceredentials削除しました！

7422767

Reply:
optionsSuccessStatus: 200削除しました！

ac493a7